megamek/src/megamek/server/Server.java

```diff
@@ -35130,6 +35159,8 @@ public void doSinkEntity(Entity en) {
 
         // get units in hex
         for (Entity entity : game.getEntitiesVector(coords)) {
+            boolean entityIsInfantryButNotBattleArmor = (entity instanceof Infantry) && !(entity instanceof BattleArmor);
```

Comment:
Why make this a variable if it is only used once?

Reply:
It's actually used twice, on line 35250 and 35263

Reply:
My search was off, my bad

Reply:
You could actually use entity.getUnitType() to do this.